src/persist-client/src/internal/machine.rs

```diff
@@ -1065,6 +1066,16 @@ where
     }
 }
 
+pub(crate) struct ExpireFn(
+    pub(crate) Box<dyn FnOnce() -> BoxFuture<'static, ()> + Send + Sync + 'static>,
+);
```

Comment:
Any specific reason we're storing a function that returns a future, and not just a future itself?

Reply:
That's what I tried first. We require that WriteHandle and ReadHandle are Send+Sync, and it turns out this Future is only Send and not Sync. Pushed up a comment.